File: experiments/beagle/anima.js
var b = require('bonescript');

var tempo = 0;
var i_pino = 0;
var pinos = [16, 21, 22, 13, 12, 11];

/* setup */
for (var i=0; i<pinos.length; i++ ) {
    var p = pinos[i];
    var pin = "P9_" + p;
    b.pinMode(pin, b.OUTPUT);
    b.digitalWrite(pin, 0);
}

function desligar() {
    b.digitalWrite("P9_" + pinos[i_pino], 0);
    i_pino++;
    if (i_pino == pinos.length) {
        i_pino = 0;
    }
    setTimeout(ligar, 0);
};

function ligar() {
    console.log(i_pino);
    b.digitalWrite("P9_" + pinos[i_pino], 1);
    setTimeout(desligar, tempo);
}

function ler() {
    b.analogRead('P9_39', function (p) {
        tempo = p.value * 1000;
    });
}

ligar();
setInterval(ler, 100);

File: experiments/beagle/anima.py
import Adafruit_BBIO.GPIO as GPIO
import Adafruit_BBIO.ADC as ADC

ADC.setup()

from time import sleep
pinos = [16, 21, 22, 13, 12, 11]

for pino in pinos:
    GPIO.setup("P9_" + str(pino), GPIO.OUT)

while True:
    for pino in pinos:
        GPIO.output("P9_" + str(pino), GPIO.HIGH)
        tempo = ADC.read('P9_39')
        print tempo
        sleep(tempo)
        GPIO.output("P9_" + str(pino), GPIO.LOW)
